mlir/lib/AsmParser/Parser.cpp

```diff
@@ -2086,11 +2089,9 @@ ParseResult OperationParser::parseLocationAlias(LocationAttr &loc) {
   Token tok = getToken();
   consumeToken(Token::hash_identifier);
   StringRef identifier = tok.getSpelling().drop_front();
-  if (identifier.contains('.')) {
-    return emitError(tok.getLoc())
-           << "expected location, but found dialect attribute: '#" << identifier
-           << "'";
-  }
+  assert(!identifier.contains('.') &&
+         "we should not be getting hash identifiers with dots here");
```

Comment:
```suggestion
         "unexpected dialect attribute token, expected alias");
```

Would be good to have the message hint to which invariant is broken

Reply:
good call